hive-metastore/src/test/java/org/apache/iceberg/hive/TestHiveCommits.java

```diff
@@ -504,6 +508,56 @@ public void testChangeLockWithAlterTable() throws Exception {
         .hasSameClassAs(initialLock);
   }
 
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  @NullSource
+  public void testFirstHiveCommitWithLockSetting(Boolean lockEnabled) {
+    String tableName =
+        Boolean.FALSE.equals(lockEnabled) ? "new_table_with_no_lock" : "new_table_with_lock";
+    TableIdentifier newTableIdentifier = TableIdentifier.of(DB_NAME, tableName);
```

Comment:
Nit: Since we drop the table anyways, we could just:
```
TableIdentifier newTableIdentifier = TableIdentifier.of(DB_NAME, "lock_test_table");
```

Reply:
Done !